java-checks/src/test/java/org/sonar/java/checks/serialization/SerializableFieldInSerializableClassCheckTest.java

```diff
@@ -39,4 +39,13 @@ void test() {
       .verifyIssues();
   }
 
+  @Test
+  void testWithoutSemantic() {
+    JavaCheckVerifier.newVerifier()
+      .onFile(nonCompilingTestSourcesPath("checks/serialization/SerializableFieldInSerializableClassCheck.java"))
```

Comment:
To answer your question, I believe we should not report Issues for lines with `@EJB` or `@Inject` if bytecode is incomplete.

There is however an easy way to cover this, we simply have to check if there is any unknown annotation. To do so, I would simply modify method `isTransientSerializableOrInjected` as follow:

```java
private static boolean isTransientSerializableOrInjected(VariableTree member) {
    if (ModifiersUtils.hasModifier(member.modifiers(), Modifier.TRANSIENT) || (isSerializable(member.type()) && !isSubtypeOfCollectionApi(member.type().symbolType()))) {
      return true;
    }
    SymbolMetadata metadata = member.symbol().metadata();
    return metadata.isAnnotatedWith("javax.inject.Inject")
      || metadata.isAnnotatedWith("javax.ejb.EJB")
      // we can not guarantee anything due to the presence of an unknown annotation
      || metadata.annotations().stream().anyMatch(annotationInstance -> annotationInstance.symbol().isUnknown());
  }
```

Reply:
Thanks. That helped me a lot :)